### ISSUE TITLE
How do CrypTen keep the shared data unrevealed to other processes on distributed servers?

### ISSUE THREAD

**Issue by TomHeaven** [facebookresearch/CrypTen#445]
I want to train an image classification model with two parties. Each of them owns a server with a part of the private image data. I do not want the private image data from one party to be known by the other party. However, I find I can decrypt cryptensors from other sources using the function "cryptensor.get_plain_text()". 

Here is my demo code. y_enc is from source 0, and z_enc is from source 1, but they can be decrypted by both processes. ``How do we keep the shared data unrevealed to other processes?``

```python
import crypten
import crypten.communicator as comm
import crypten.mpc as mpc
import os

os.environ['CUDA_VISIBLE_DEVICES'] = ''


crypten.init()


@mpc.run_multiprocess(world_size=2)
def examine_arithmetic_shares():
    rank = comm.get().get_rank()

    y_enc = crypten.cryptensor(
        [1, 20, 10], ptype=crypten.mpc.arithmetic, src=0)
    z_enc = crypten.cryptensor(y_enc.get_plain_text(), src=1)
    print("Rank ", rank, 'y plain text', y_enc.get_plain_text(),
          ' z plain text', z_enc.get_plain_text())


examine_arithmetic_shares()
```

**Comment by mohammad-alrubaie** [comment]
The function you have is a shared program that is run by both parties, so in this scenario, it is agreed upon by all parties to release the plain text values.
If instead, you run two processes, and each one has a different code, with one using get_plain_text and the other doesn't, then it should be the case that that party's process will halt. See https://github.com/facebookresearch/CrypTen/issues/233
If all parties share the same program, then the program will run without issues, and if they all agree to release part of the data to one party only using `get_plain_text(dst=destination_party)`, then that should be fine (see https://github.com/facebookresearch/CrypTen/issues/358). 
To summarize, it is built-in behavior to prevent release when some parties refuse to share.

**Comment by TomHeaven** [comment]
Thank you for the explanation. Maybe we can add it to the documentation. I believe other users may have the some doubt.

**Comment by TomHeaven** [comment]
My distributed demo is available at: [https://github.com/TomHeaven/crypten_cifar10_cuda_distributed](https://github.com/TomHeaven/crypten_cifar10_cuda_distributed).
